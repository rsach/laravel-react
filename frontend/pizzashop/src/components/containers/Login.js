import { connect } from 'react-redux'
import { login} from '../../actions'


import Login from "../ui/Login";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onLogin(body){
            dispatch(
                login(body)
            )
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Login)
