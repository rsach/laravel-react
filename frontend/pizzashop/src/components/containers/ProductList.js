import { connect } from 'react-redux'
import {addToCart, fetchProducts} from '../../actions'
import ProductList from "../ui/ProductList";

const mapStateToProps = (state,props) =>
    ({
        products:state.products.products,
    });

const mapDispatchToProps = dispatch =>
    ({
        onAddToCart(product){
            dispatch(
                addToCart(product)
            )
        },
        fetchProducts() {
            dispatch(
                fetchProducts('')
            )
        }
    });

export default connect(mapStateToProps,mapDispatchToProps)(ProductList)
