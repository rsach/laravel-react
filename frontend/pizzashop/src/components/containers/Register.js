import { connect } from 'react-redux'
import { register} from '../../actions'


import Register from "../ui/Register";

const mapStateToProps = (state,props) =>
    ({

    });

const mapDispatchToProps = dispatch =>
    ({
        onSignUp(body){
            console.log(body)
            dispatch(
                register(body)
            )

            return body
        }

    });

export default connect(mapStateToProps,mapDispatchToProps)(Register)
