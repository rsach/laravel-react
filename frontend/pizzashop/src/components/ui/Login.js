import React, {Component} from 'react';
import PropTypes from "prop-types";
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
// import AppBar from 'material-ui/AppBar';
import AppBar from '@material-ui/core/AppBar';



import history from "../../history";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";

class Login extends Component {
    constructor(props) {
        super(props);

        // reset login status


        this.state = {
            email: '',
            password: '',
            submitted: false
        };


        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }




    handleSubmit(e) {
        e.preventDefault();

        this.setState({ submitted: true });
        const { email, password } = this.state;
        const {onLogin=f=>f} = this.props;

        if (email && password) {

            onLogin({email, password});
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/register')
    }

    render() {


        return (

            <div>
                <MuiThemeProvider>
                    <div>
                        <AppBar
                            title="Login"
                        />
                        <div className="flex  flex-center">
                            <div className="form flex flex-column flex-center">


                                <TextField
                                    id="email"
                                    data-testid="email"
                                    variant="outlined"
                                    placeholder="Enter your Email"
                                    label="Email"
                                    onChange = {(event) => this.setState({email:event.target.value})}
                                />
                                <br/>
                                <TextField
                                    data-testid="password"

                                    id="password"
                                    variant="outlined"
                                    type="password"
                                    placeholder="Enter your Password"
                                    label="Password"
                                    onChange = {(event) => this.setState({password:event.target.value})}
                                />
                                <br/>
                                <div className="flex button-margin flex-space-between">
                                    <Button                    id="login"                 data-testid="login"
                                                                                variant="contained" color="primary"  onClick={(event) => this.handleSubmit(event)}>Login</Button>

                                        <Button id="register" variant="contained"        data-testid="register"
                                                onClick={(e) => this.handleRedirect(e)}  color="secondary"    >
                                            Register
                                        </Button>




                                </div>
                            </div>

                        </div>

                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
}


Login.propTypes = {
    onLogin: PropTypes.func.isRequired
};

export default Login

