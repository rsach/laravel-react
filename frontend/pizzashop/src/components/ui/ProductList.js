
import '../../stylesheets/index.scss'
import Product from "../containers/Product";
import React from 'react'
import AppBar from "material-ui/AppBar";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

const ProductList = ({products = []}) =>
    (
                <div>
                    <MuiThemeProvider>
                        <div>
                            <AppBar
                               className="fixed" title="Pizza List"
                            />

                            <div className="product-list flex flex-wrap">

                                {
                                    (products || []).map((product,i) => <Product key={i} product={product} />)
                                }

                            </div>

                        </div>
                    </MuiThemeProvider>
                </div>

            );




 

ProductList.propTypes = {
    products: (props) => (!Array.isArray(props.products)) ?
        new Error("Product List property must be an array") :
            null,


};

export default ProductList