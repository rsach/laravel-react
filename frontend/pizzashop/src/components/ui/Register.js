import React, {Component} from 'react';

import PropTypes from "prop-types";


import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';


import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import history from "../../history";


class Register extends Component {
    constructor(props) {
        super(props);

        this.state = {
            user: {
                name: '',
                email: '',
                password: ''
            },
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleRedirect = this.handleRedirect.bind(this);
    }

    handleChange(newValue) {
        // const { name, value } = event;
        const { user } = this.state;
        this.setState({
            user: {
                ...user,
                ...newValue
            }
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        this.setState({ submitted: true });
        const { user } = this.state;
        const {onSignUp=f=>f} = this.props;

        if (user.name && user.email && user.password) {

            onSignUp(user);
        }
    }

    handleRedirect(e) {
        e.preventDefault();
        history.push('/login')
    }

    render() {

        // const { user, submitted } = this.state;
        return (
            <div>
                <MuiThemeProvider>
                <div>
                    <AppBar
                        title="Register"
                    />
                    <div className="flex  flex-center">
                        <div className="form flex flex-column">

                                <TextField
                                    id="name"
                                    variant="outlined"

                                placeholder="Enter your Name"
                                label="Name"
                                onChange = {(event) => this.handleChange({name: event.target.value})}
                            />


                            <br/>
                            <TextField
                                id="email"
                                placeholder="Enter your Email"
                                type="email"
                                variant="outlined"

                                label="Email"
                                onChange = {(event) => this.handleChange({email: event.target.value})}
                            />
                            <br/>
                            <TextField
                                id="password"
                                type = "password"
                                variant="outlined"

                                placeholder="Enter your Password"
                                label="Password"
                                onChange = {(event) => this.handleChange({password: event.target.value})}
                            />
                            <br/>
                            <div className="flex button-margin flex-space-between">
                                <Button id="register" label="Register" variant="contained"  color="primary"  onClick={(event) => this.handleSubmit(event)}> Register </Button>


                                    <Button id="login" variant="contained" label="Login" color="secondary" onClick={(e) => this.handleRedirect(e)}    >

                                        Login
                                    </Button>



                            </div>
                        </div>
                    </div>
                </div>
            </MuiThemeProvider>
            </div>
        );
    }
}


Register.propTypes = {
    onSignUp: PropTypes.func,
};

export default Register
