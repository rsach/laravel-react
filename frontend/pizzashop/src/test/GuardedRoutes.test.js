import React from 'react';
import { render } from '@testing-library/react';
import Login from "../components/ui/Login";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Register from "../components/ui/Register";
import ProductList from "../components/ui/ProductList";
import PropTypes from "prop-types";
import P, {PrivateRoute} from '../guardedRoute';
import {MemoryRouter, Route} from "react-router";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";


jest.mock('')
Enzyme.configure({ adapter: new Adapter() });

const localStorageMock = (() => {
    let store = {};

    return {
        getItem(key) {
            return store[key] || null;
        },
        setItem(key, value) {
            store[key] = value.toString();
        },
        removeItem(key) {
            delete store[key];
        },
        clear() {
            store = {};
        }
    };
})();

Object.defineProperty(window, 'localStorage', {
    value: localStorageMock
});


Object.defineProperty(window, 'localStorage', {
    value: localStorageMock
});



const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<PrivateRoute/> spec', () => {


    function setup(prop) {
        const props = {
            children: () => <div id="success">gh</div>,
            auth_guard: false,
            ...prop
        };
        const store = mockStore({});

        const wrapper = shallow(

                <PrivateRoute {...props} />

    );
        return {
            props,  wrapper, store
        }
    }







    it('renders the component', () => {
        const {wrapper} = setup();
        expect(wrapper.firstChild).toMatchSnapshot();
    });

    it('renders the component', () => {

        const {wrapper} = setup({path: '/'});
        expect(wrapper.find(Route).prop('path')).toBe('/');
    });


    it('renders the component', () => {

        const { wrapper, props } = setup({ auth_guard: true });
        const ComponentToRender = wrapper.prop('render');
        const componentWrapper = shallow(<ComponentToRender location="/scarif" />);
        expect(componentWrapper.find('#success').length).toBe(1);

    });

    it('renders the component', () => {

        const { wrapper, props } = setup({ auth_guard: false });
        const ComponentToRender = wrapper.prop('render');
        const componentWrapper = shallow(<ComponentToRender location="/scarif" />);
        expect(componentWrapper.find('#fail').length).toBe(1);

    });

    it('renders the component', () => {

        let { wrapper, props, store } = setup({ auth_guard: true });
        wrapper = shallow(
            <Provider store={store}>
                <P  {...props}></P>
            </Provider>
        );

        expect(wrapper.firstChild).toMatchSnapshot();



    });





});