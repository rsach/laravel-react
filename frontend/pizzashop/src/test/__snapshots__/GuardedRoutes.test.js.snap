// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute/> spec renders the component 1`] = `undefined`;

exports[`<PrivateRoute/> spec renders the component 2`] = `undefined`;
